### PR TITLE
gh-456: call `redshift_grid` within `distance_grid`

### LINKED ISSUE #456: `shells.distance_grid` should call `shells.redshift_grid`
### Is Your Feature Request Related to a Problem? Please Describe

The contents of `shells.redshift_grid` are entirely contained within `shells.distance_grid`, so it makes sense to combine them.

### Describe the Solution You'd Like

_No response_

### Describe Alternatives You've Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by paddyroddy**
Closes #456. Discovered in #449. The contents of the two functions are near identical so makes sense to combine.

### CODE REVIEW

**Comment by ntessore** [comment]
Hm, I think I'm -1 on this. No harm in duplicating 10 lines of code, but a lot of harm in people potentially misunderstanding what the distance grid does.

**Comment by ntessore** [comment]
We could implement both in terms of a private `_uniform_grid()` function maybe?

File: glass/shells.py [hunk 1]
```diff
@@ -708,15 +708,15 @@ def partition_restrict(
     return part
 
 
-def redshift_grid(
+def _uniform_grid(
     zmin: float,
     zmax: float,
     *,
     dz: float | None = None,
     num: int | None = None,
```
Comment (ntessore) [inline glass/shells.py:716]: Maybe something more generic here?

```suggestion
    start: float,
    stop: float,
    *,
    step: float | None = None,
    num: int | None = None,
```

File: glass/shells.py [hunk 2]
```diff
@@ -708,15 +708,15 @@ def partition_restrict(
...
 ) -> npt.NDArray[np.float64]:
     """
-    Redshift grid with uniform spacing in redshift.
+    Create a redshift grid.
```
Comment (ntessore) [inline glass/shells.py:719]:
```suggestion
    Create a uniform grid.
```

File: glass/shells.py [hunk 3]
```diff
@@ -744,11 +744,40 @@ def redshift_grid(
     elif dz is None and num is not None:
         z = np.linspace(zmin, zmax, num + 1)
     else:
-        msg = 'exactly one of "dz" or "num" must be given'
+        msg = "exactly one of 'dz' or 'num' must be given"
```
Comment (ntessore) [inline glass/shells.py:747]:
```suggestion
        msg = "exactly one of grid step size or number of steps must be given"
```

File: glass/shells.py (hunk 1 above)
Comment (paddyroddy) [reply glass/shells.py:716]: Good idea

File: glass/shells.py [hunk 4]
```diff
@@ -744,11 +744,40 @@ def redshift_grid(
...
+        The redshift spacing.
+    num
+        The number redshift samples.
+
+    Returns
+    -------
+        The redshift grid.
+
+    """
+    return _uniform_grid(zmin, zmax, dz=dz, num=num)
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply glass/shells.py:778]: Yeah I know, haven't finished yet

**Review by Saransh-cpp — APPROVED** [review]
Nice, looks good!

**Review by ntessore — APPROVED** [review]
Nice catch! Good to not have to implement the logic twice.

**Pull request merged**